### ISSUE TITLE
training tfd.TransformedDistribution decorated with @tf.function throws error 

### ISSUE THREAD

**Issue by gitlabspy** [tensorflow/probability#1094]
As long as I decorate my train step function with `@tf.function`, it throws the error below and it takes so long to start training.
```
'chain_of_batch_normalization_of_ffjord_of_batch_normaliza
tion_of_ffjord_of_batch_normalization_of_ffjord_of_batch_normalization_of_ffjord_of_batch_normalization_of_ffjord_of_batch_normalizati
on_of_ffjordreshapeMultivariateNormalDiag/log_prob/chain_of_batch_normalization_of_ffjord_of_batch_normalization_of_ffjord_of_batch_no
rmalization_of_ffjord_of_batch_normalization_of_ffjord_of_batch_normalization_of_ffjord_of_batch_normalization_of_ffjord/inverse/ffjor
d_5/inverse/dormand_prince/while/next_iteration/_2989', 

.... THERE'S A LOT OF THEM .....

'dormand_princeGradients/while/body/_3858/dormand_princeGradients/while/dormand_prince/while/body/
_28854/dormand_princeGradients/while/dormand_prince/while/advance_to_solution_time/while/LoopExecuted/_62154', 'dormand_princeGradient
s/while/body/_3858/dormand_princeGradients/while/dormand_prince/while/body/_28854/dormand_princeGradients/while/dormand_prince/while/I
dentity_66' -> 'dormand_princeGradients/while/body/_3858/dormand_princeGradients/while/dormand_prince/while/next_iteration/_29998', 'F
unc/dormand_princeGradients/while/body/_3858/gradient_tape/dormand_princeGradients/while/map/while/dormand_princeGradients/while/map/w
hile_grad/body/_28379/input/_42770' -> 'dormand_princeGradients/while/body/_3858/gradient_tape/dormand_princeGradients/while/map/while
/dormand_princeGradients/while/map/while_grad/body/_28379/gradient_tape/dormand_princeGradients/while/map/while/gradients/AddN_21'}.
```
 And it takes so long to pop up these two lines:
```
2020-09-30 16:50:35.084339: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libc
udnn.so.7
2020-09-30 16:50:43.687273: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcub
las.so.10
```
My train step function looks like:
`stacked_ffjord` is a chain which contains some tfb.ffjord (which I modified a bit for passing kwargs) and tfb.BatchNormalization.

```
base_loc = np.zeros(np.prod((36, 60, 3))).astype(np.float32)
base_sigma = np.ones(np.prod((36, 60, 3))).astype(np.float32)
base_distribution = tfd.MultivariateNormalDiag(base_loc, base_sigma)
base_distribution = tfd.TransformedDistribution(distribution=base_distribution, bijector=tfb.Reshape((36, 60, 3)))

transformed_distribution = tfd.TransformedDistribution(
    distribution=base_distribution, bijector=stacked_ffjord) 
@tf.function
def train_step(optimizer, target_sample, condition):
    with tf.GradientTape() as tape:
        loss = -tf.reduce_mean(transformed_distribution.log_prob(target_sample, **{'bijector_kwargs':{'ffjord':{'condition':condition}}}))
    variables = tape.watched_variables()
    gradients = tape.gradient(loss, variables)
    optimizer.apply_gradients(zip(gradients, variables))
    return loss

```

Training process goes well after these two popped up but I wonder how can I avoid these error/warning and make it faster to start training...
